### ISSUE TITLE
Improve documentation around conflicts

### ISSUE THREAD

**Issue by Neppord** [tree-sitter/tree-sitter#654]
It took me some time to realize that the `conflicts` member of the grammar needs to be a function that returns a array of arrays. in the documentation it says it should be a array of arrays.

**Comment by maxbrunsfeld** [comment]
Yeah, this is true for all of the fields of the grammar besides `rules`. Maybe we should just add something in [the introductory paragraph](https://tree-sitter.github.io/tree-sitter/creating-parsers#the-grammar-dsl) of the overview of the DSL, saying that all of the other fields are functions (so that you can have the `$` object as a parameter).

**Comment by Neppord** [comment]
better to add it to this paragraph then
> In addition to the name and rules fields, grammars have a few other optional public fields that influence the behavior of the parser

There is also a inconsistency there it is shown in the `extras` section, even with a example. But not of the others get this luxery. There are also examples of most of the other fields but not `conflicts`.

**Comment by panlina** [comment]
+1. If precedence and associativity both have examples, conflicts should have too. It will be helpful for users that do not know what's a "LR(1) conflict". I don't know how to use it even by looking at examples in other parser grammars.
